processing/src/main/java/io/druid/segment/filter/DimensionPredicateFilter.java

```diff
@@ -19,36 +19,45 @@
 
 package io.druid.segment.filter;
 
-import com.google.common.base.Function;
 import com.google.common.base.Predicate;
 import com.metamx.collections.bitmap.ImmutableBitmap;
-import com.metamx.common.guava.FunctionalIterable;
 import io.druid.query.extraction.ExtractionFn;
 import io.druid.query.filter.BitmapIndexSelector;
 import io.druid.query.filter.Filter;
 import io.druid.query.filter.ValueMatcher;
 import io.druid.query.filter.ValueMatcherFactory;
+import io.druid.segment.column.BitmapIndex;
 import io.druid.segment.data.Indexed;
 
-import javax.annotation.Nullable;
+import java.util.Iterator;
 
 /**
  */
-class DimensionPredicateFilter implements Filter
+public class DimensionPredicateFilter implements Filter
 {
   private final String dimension;
   private final Predicate<String> predicate;
-  private final ExtractionFn extractionFn;
 
   public DimensionPredicateFilter(
-      String dimension,
-      Predicate<String> predicate,
-      ExtractionFn extractionFn
+      final String dimension,
+      final Predicate<String> predicate,
+      final ExtractionFn extractionFn
   )
   {
     this.dimension = dimension;
-    this.predicate = predicate;
-    this.extractionFn = extractionFn;
+
+    if (predicate == null) {
+      this.predicate = null;
+    } else {
+      this.predicate = extractionFn == null ? predicate : new Predicate<String>()
+      {
+        @Override
+        public boolean apply(String input)
+        {
+          return predicate.apply(extractionFn.apply(input));
```

Comment:
I'm not really sure, since I don't know why `predicate` is allowed to be null, or when that would happen. I was thinking about putting a checkNotNull in there but I wanted to avoid changing behavior.